### ISSUE TITLE
Snoozed job cannot be viewed from UI

### ISSUE THREAD

**Issue by cfelegy** [riverqueue/riverui#103]
When viewing a job that I snoozed (returned `river.JobSnooze(time.Duration)`), the UI just displays a blank screen. 

I am running the latest published container, `docker run -p 3001:8080 --env DATABASE_URL ghcr.io/riverqueue/riverui:latest`

The stacktrace suggests:
```
TypeError: Cannot read properties of undefined (reading 'at')
    at sk (index-CLiUj5sP.js:207:71592)
    at mo (react-dom-BOasIFCf.js:30:16959)
    at Ha (react-dom-BOasIFCf.js:32:43694)
    at Va (react-dom-BOasIFCf.js:32:39499)

const t = e.finalizedAt || e.errors[e.errors.length - 1].at
      , n = e.state === N.Completed;
```

Now, I'm not sure if this issue falls under the UI project or river itself. Checking the `river_job` table directly showed that the `errors` object for the snoozed job is empty, so it makes sense that `e.errors[].at` would fail. The docs seem to suggest that a snoozed job is effectively just re-scheduled for the future without being counted as an error, so I guess it's logical that the errors array would remain empty here.  However, the finalized timestamp also remains null, since the job didn't run to completion.

The offending line in the frontend appears to be here, if I mentally de-minified correctly: https://github.com/riverqueue/riverui/blob/master/ui/src/components/JobTimeline.tsx#L234

A suitable patch could be checking if the job is scheduled before that line, and returning a StatusStep with name "snoozed". 

<details><summary>Screenshots</summary>
<p>
<img width="1800" alt="image" src="https://github.com/user-attachments/assets/caa0531d-6717-4477-b4bf-4e7698d9c7a4">
<img width="875" alt="image" src="https://github.com/user-attachments/assets/ef005c05-417e-45b1-920d-0870db81233b">
</p>
</details>

**Comment by bgentry** [comment]
Thanks for the report, will get this sorted shortly!

**Comment by bgentry** [comment]
Should be resolved by #104 as soon as that's merged and released 🙏

**Comment by brandur** [comment]
We just cut v0.3.1 which includes @bgentry's fix.
